Initial Honeycomb Integration

HoloRin opened the pull request:
## Proposed Changes

Collect test statistics with Honeycomb.io. Captures 

- Github Workflow
- Github Run Id
- Github Repository
- Github Sha
- Github Ref
- Test Suite Name
- Test Case Name
- Test Case Duration
- Test Case Result

only at this time, but this should be enough to perform some queries in Honeycomb to identify flaky and/or slow tests.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [x] Continuous Integration

HoloRin committed:
Run test suite via GitHub Actions on every push

Generated with `make github-actions`

HoloRin committed:
Dedupe calls to rabbit_mgmt_test_util:reset_management_setting/1

It was called as a setup step in init_per_group, but also in every init_per_testcase.

pjk25 commented:
Only the last commit matters, as this builds off of #809 which is not merged yet.

HoloRin committed:
Collect test results in honeycomb.io

Initial spike with test results and timing. The intention is to be able to find flaky tests statistically.

HoloRin committed:
Send honeycomb json files to a local dir

Rather than /tmp/...

HoloRin committed:
Add additional data to honeycomb events

HoloRin committed:
Fix workflow error

HoloRin committed:
Update the honeycomb dataset, and regenerate the workflows

pjk25 commented on .github/workflows/test-jobs/30-CT_SUITE.yaml:
```diff
@@ -0,0 +1,77 @@
+      - name: HONEYCOMB
+        run: |
+          for f in honeycomb/*; do
+            curl -H 'X-Honeycomb-Team: ${{ secrets.HONEYCOMB_TEAM }}' \
```
This should receive the same return code validation as was added to rabbitmq-cli, and probably also move to the tail of the steps, so that if it fails, it does not affect calls to `success()` in the `if:` of the mixed versions tests.
